### PR TITLE
Use the stateful midi-renderer

### PR DESCRIPTION
**Pull request by mendelt**
This PR implements the MidiTransport on top of an embedded serial Writer and uses that to implement a midi interface on top of the midi-convert MidiRenderer.

This is implemented on top of a midi-convert branch so that needs to be merged first before this can be merged.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -39,9 +42,24 @@ where
...
+    TX: serial::Write<u8, Error = E>,
+    E: Debug,
+{
+    type Error = E;
+
+    fn write(&mut self, value: u8) -> Result<(), Self::Error> {
+        block!(self.0.write(value))
+    }
+}
+
```
Comment (x37v) [inline src/lib.rs]: maybe just a preference but, i don't believe you need this struct at all:

```
impl<TX, E> MidiTransport for TX
where
    TX: serial::Write<u8, Error = E>,
    E: Debug,
{
    type Error = E;

    fn write(&mut self, value: u8) -> Result<(), Self::Error> {
        block!(self.write(value))
    }
}
```

Comment (mendelt) [reply src/lib.rs]: That would be my preference too, and I tried this first. But you can only implement traits when the trait or the type that implements it is in your own crate. In this case serial::Write is from the embedded-hal crate and MidiTransport is from the midi-convert crate. So rust didnt let me do that :-( .

Comment (x37v) [reply src/lib.rs]: Ahh yeah, I forgot about that!

**Review by x37v — COMMENTED** [review]
love removing code! looks good, I assume we can just do a `0.2.0` release of `midi-convert` and update to that instead of the git soon?

**Pull request merged**